docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
+* supporting the parameterization of charts, as opposed to templates.
+* supporting more than one top level template in a single chart.
+* supporting dependency definition and resolution mechanism at the chart level.
+
+It was also observed that there is strong motivation to complete the first pass of the helm/dm integration, using the existing DM templating engine, before adding new features or addressing more complex scenarios.
+
+## Assertions
+In view of this context, commentators asserted that:
+* It seems unlikely for a single chart to require more than one generation technology.
+* It seems unnecessary that we need to support more than one top level template per chart, if templates can nest, since any forest of top level templates required by the user can be implemented using a tree, where a single top level template calls the members of the forest.
```

Comment:
Agreed. Might be worth writing a simple Go template iterator that takes a block of input parameters and then expands a set of Go templates, passing the appropriate parameters to each template. Would require a way to map parameters to templates, such as qualification (e.g., if we have two templates, a and b, then instead of simple names like foo and bar, parameters have qualified name, such as a/foo and b/bar). However we do it, it's still pushing the problem to the template get engine, instead of trying to solve it at the chart level, which is the only workable approach, IMO.